pkgs/build-support/kernel/make-initrd-ng.nix

```diff
@@ -78,14 +78,14 @@ in
 
   STRIP = if strip then "${pkgsBuildHost.binutils.targetPrefix}strip" else null;
 }) ''
-  mkdir ./root
+  mkdir -p ./root/var/empty
   make-initrd-ng "$contentsPath" ./root
   mkdir "$out"
   (cd root && find * .[^.*] -exec touch -h -d '@1' '{}' +)
   for PREP in $prepend; do
     cat $PREP >> $out/initrd
   done
-  (cd root && find * .[^.*] -print0 | sort -z | cpio -o -H newc -R +0:+0 --reproducible --null | eval -- $compress >> "$out/initrd")
+  (cd root && find . -print0 | sort -z | cpio -o -H newc -R +0:+0 --reproducible --null | eval -- $compress >> "$out/initrd")
```

Comment:
This looks like a simplification to me so I'm just wondering whether it is intended that the new command includes an extra output `.` compared to the replaced command (no `-mindepth 1`).

Also [this](https://unix.stackexchange.com/questions/453127) might be an interesting related note. Not sure if applicable here.

Reply:
> wondering whether it is intended that the new command includes an extra output `.` compared to the replaced command

Including the root directory in the `find` output is important because it ensures it has the proper mode in the initrd.

> Also [this](https://unix.stackexchange.com/questions/453127) might be an interesting related note. Not sure if applicable here.

I don't believe that's applicable. We aren't unpacking this archive at all; the kernel just reads it into memory at startup. Plus one of the comments in that thread suggests most cpio implementations don't have this problem, including the ones installed on NixOS and macOS.